edb/server/protocol/execute.pyx

```diff
@@ -358,28 +344,16 @@ async def execute(
                 #   1. An orphan ROLLBACK command without a paring start tx
                 #   2. There was no SQL, so the state can't have been synced.
                 be_conn.last_state = state
+        if compiled.recompiled_cache:
+            for req, qu_group in compiled.recompiled_cache:
+                dbv.cache_compiled_query(req, qu_group)
         if (
-            debug.flags.persistent_cache
-            and not dbv.in_tx()
-            and not query_unit.tx_rollback
-            and query_unit.user_schema
-            and server.config_lookup(
-                "auto_rebuild_query_cache",
-                dbv.get_session_config(),
-                dbv.get_database_config(),
-                dbv.get_system_config(),
-            )
+            not debug.flags.disable_persistent_cache
+            and compiled.recompiled_cache
+            or compiled.request
+            and query_unit.cache_sql
```

Comment:
Add extra parens to separate the `and` and `or` clauses. I can never remember their precedence so I can't understand it really

Reply:
(OK I'm pretty confident that `and` binds tighter because `and` is like multiplication and is sometimes in logic written as juxtaposition, but I still think we should avoid it.)

Reply:
Extracted 2 variables and added more comments :v: